ykrt/src/mt.rs

```diff
@@ -283,25 +295,40 @@ impl MTThread {
                     };
 
                     if stop_tracer {
-                        if pack.compare_and_swap(lp, PHASE_COMPILED, Ordering::Release) == lp {
-                            let _sir_trace = Rc::get_mut(&mut self.inner)
-                                .unwrap()
-                                .tracer
-                                .take()
-                                .unwrap()
-                                .0
-                                .stop_tracing();
-                            // FIXME build TIR and compile. Eventually in a background thread?
+                        let sir_trace = Rc::get_mut(&mut self.inner)
+                            .unwrap()
+                            .tracer
+                            .take()
+                            .unwrap()
+                            .0
+                            .stop_tracing()
+                            .unwrap();
+                        let tir_trace = TirTrace::new(&*SIR, &*sir_trace).unwrap();
+                        let ct = TraceCompiler::<I>::compile(tir_trace);
+                        let ptr = Box::into_raw(Box::new(ct)) as usize; //ct.ptr() as usize;
+
+                        let new_pack = PHASE_COMPILED | (ptr << PHASE_NUM_BITS);
+                        if pack.compare_and_swap(lp, new_pack, Ordering::Release) == lp {
```

Comment:
At this point I think we know we're the only thread that can try writing to this, in which case I think `pack.store` w/ `Ordering::Release` is sufficient (i.e. I don't think the CAS does anything useful apart from doing a store).

Reply:
Will try.

Reply:
Fixed in bc9291b.